src/guide/blockchain/metadata.md

```diff
@@ -0,0 +1,162 @@
+# Metadata
+
+Metadata are key-value pairs that are attached to objects in the
+blockchain. The following can contain metadata:
+
+- domains
+- accounts
+- assets
+- asset definitions
+- triggers
+- transactions
+
+The metadata can be of very different types, such as:
+
+- structures with named or unnamed fields
+- enums
+- integers
+- numbers with fixed decimal precision
+- strings
+- Boolean values
+- arrays
+- associative arrays
+- vectors
+- request results
```

Comment:
Metadata maps keys to JSON strings: https://github.com/hyperledger/iroha/issues/4353